applications/GeoMechanicsApplication/custom_processes/apply_normal_load_table_process.cpp

```diff
@@ -0,0 +1,195 @@
+                                                                      NamesOfSettingsToCopy);
+    normal_parameters.AddValue("value", rProcessSettings["value"][0]);
+
+    if (FunctonsForParameters::HasTableAttached(rProcessSettings,0)) {
+        normal_parameters.AddValue("table", rProcessSettings["table"][0]);
+        mProcesses.push_back(std::make_unique <ApplyComponentTableProcess>(mrModelPart, 
+                                                                           normal_parameters));
+    } else {
+        mProcesses.push_back(std::make_unique <ApplyConstantScalarValueProcess>(mrModelPart, 
+                                                                                normal_parameters));
+    }
+}
+
+void ApplyNormalLoadTableProcess::MakeProcessForHydrostaticFluidPressureType(const Parameters& rProcessSettings,
+                                                                             std::vector<std::string> NamesOfSettingsToCopy)
+{
+    NamesOfSettingsToCopy.insert(NamesOfSettingsToCopy.end(), { "gravity_direction",
+                                                               "reference_coordinate",
+                                                               "specific_weight" });
+    if (FunctonsForParameters::HasTableAttached(rProcessSettings, 0)) {
+        auto normal_parameters = FunctonsForParameters::ExtractParameters(rProcessSettings,
+                                                                          NamesOfSettingsToCopy);
+        normal_parameters.AddValue("table", rProcessSettings["table"][0]);
+        mProcesses.push_back(std::make_unique <ApplyBoundaryHydrostaticPressureTableProcess>(mrModelPart, 
+                                                                                             normal_parameters));
+    }
+    else {
+        mProcesses.push_back(std::make_unique <ApplyConstantBoundaryHydrostaticPressureProcess>(mrModelPart, 
+                                                                                                FunctonsForParameters::ExtractParameters(rProcessSettings,NamesOfSettingsToCopy)));
+    }
+}
+
+void ApplyNormalLoadTableProcess::MakeProcessForPhreaticLineFluidPressureType(const Parameters& rProcessSettings,
+                                                                              std::vector<std::string> NamesOfSettingsToCopy)
+{
+    NamesOfSettingsToCopy.insert(NamesOfSettingsToCopy.end(), { "gravity_direction",
+                                                               "out_of_plane_direction",
+                                                               "first_reference_coordinate",
+                                                               "second_reference_coordinate",
+                                                               "specific_weight"});
+    if (FunctonsForParameters::HasTableAttached(rProcessSettings)) {
+        NamesOfSettingsToCopy.insert(NamesOfSettingsToCopy.end(), { "table" });
```

Comment:
Although correct, I would simplify this to:
```suggestion
        NamesOfSettingsToCopy.emplace_back("table");
```

Reply:
Thank you. It is modified now.